components/script/dom/bindings/utils.rs

```diff
@@ -42,34 +43,29 @@ use crate::dom::bindings::error::throw_invalid_this;
 use crate::dom::bindings::inheritance::TopTypeId;
 use crate::dom::bindings::str::DOMString;
 use crate::dom::bindings::trace::trace_object;
-use crate::dom::windowproxy;
+use crate::dom::windowproxy::{WindowProxyHandler, PROXY_HANDLER};
 use crate::script_runtime::JSContext as SafeJSContext;
 
-/// Proxy handler for a WindowProxy.
-pub struct WindowProxyHandler(pub *const libc::c_void);
-
-impl MallocSizeOf for WindowProxyHandler {
-    fn size_of(&self, _ops: &mut MallocSizeOfOps) -> usize {
-        // FIXME(#6907) this is a pointer to memory allocated by `new` in NewProxyHandler in rust-mozjs.
-        0
-    }
-}
-
-#[derive(JSTraceable, MallocSizeOf)]
+#[derive(JSTraceable)]
 /// Static data associated with a global object.
 pub struct GlobalStaticData {
     /// The WindowProxy proxy handler for this global.
-    pub windowproxy_handler: WindowProxyHandler,
+    pub windowproxy_handler: Arc<WindowProxyHandler>,
```

Comment:
Why Arc instead of Rc?

Reply:
Didn't try `Rc` but can try it out. I just saw that `Arc` was used elsewhere in a `Window` struct member and assumed it would be required.

Reply:
Ok, fixed.